### ISSUE TITLE
Refactor handling exceptions in base fuzzing flow method

### ISSUE THREAD

**Issue by EgorkaKulikov** [UnitTestBot/UTBotJava#658]
**Description**

Current implementation of base fuzzing flow method contains the following code:

```kotlin
catch (e: CancellationException) {
            logger.debug { "Cancelled by timeout" }
        } catch (e: ConcreteExecutionFailureException) {
            emitFailedConcreteExecutionResult(initialEnvironmentModels, e)
        } catch (e: Throwable) {
            emit(UtError("Default concrete execution failed", e))
        }
```

There are several problems:
1) catching `Throwable` looks like a black hole to hide real problems
2) it is unclear why any throwable in the last catch block relates to`ConcreteExecutor`
3) A message `default concrete execution failed` is absolutely unclear either for the user or for the developer

**Comment by alisevych** [comment]
Mentioned in Manual testing task by mistake

**Comment by Markoutte** [comment]
Fixed with #663
